lib/scala/project-manager/src/main/scala/org/enso/projectmanager/util/Logging.scala

```diff
@@ -0,0 +1,67 @@
+package org.enso.projectmanager.util
+
+import ch.qos.logback.classic.{Level, LoggerContext}
+import org.slf4j.LoggerFactory
+
+import scala.util.Try
+
+object Logging {
+
+  /** Application log level. */
+  sealed trait LogLevel
+  object LogLevel {
+
+    case object Error   extends LogLevel
+    case object Warning extends LogLevel
+    case object Info    extends LogLevel
+    case object Debug   extends LogLevel
+    case object Trace   extends LogLevel
+
+    /** Convert to logback log level. */
+    def toLogback(level: LogLevel): Level =
+      level match {
+        case Error   => Level.ERROR
+        case Warning => Level.WARN
+        case Info    => Level.INFO
+        case Debug   => Level.DEBUG
+        case Trace   => Level.TRACE
+      }
+
+    /** Convert from logback log level. */
+    def fromLogback(level: Level): LogLevel = {
+      level match {
+        case Level.`ERROR` => Error
+        case Level.`WARN`  => Warning
+        case Level.`INFO`  => Info
+        case Level.`DEBUG` => Debug
+        case Level.`TRACE` => Trace
+      }
+    }
+  }
+
+  private val ROOT_LOGGER = "org.enso"
+
+  /**
+    * Set log level for the application root logger.
+    *
+    * @param level the log level
+    * @return the new log level
+    */
+  def setLogLevel(level: LogLevel): Option[LogLevel] = {
```

Comment:
What is the rationale of returning `Option` here and in the getter? It seems just returning the `Try` would carry more information (in the callsite, if setting log level fails, the application can report what was the cause of failure).

Reply:
I'll check what is the error message reported. If it is readable, I'll print it as well :+1: